### PR TITLE
Fix more authn jwt comments 3

### PR DESCRIPTION
**Pull request by tzheleznyak**
Fixes comments from the big authn-jwt PR
https://github.com/cyberark/conjur/pull/2227

### CODE REVIEW

File: app/domain/authentication/authn_jwt/signing_key/create_jwks_from_http_response.rb [hunk 1]
```diff
@@ -20,32 +21,38 @@ def validate_response_exists
...
+            keys = parsed_response['keys']
+          rescue => e
+            raise Errors::Authentication::AuthnJwt::FailedToConvertResponseToJwks.new(
+              encoded_body,
+              e.inspect
+            )
+          end
+
+          validate_keys_not_empty(keys, encoded_body)
+          return [{ keys: @jwk_set_class.new(keys) }, keys]
```
Comment (orenbm) [inline app/domain/authentication/authn_jwt/signing_key/create_jwks_from_http_response.rb:51]: you don't need to return the keys now
```suggestion
          return { keys: @jwk_set_class.new(keys) }, keys]
```

File: app/domain/authentication/authn_jwt/signing_key/fetch_jwks_uri_signing_key.rb [hunk 2]
```diff
@@ -17,12 +17,12 @@ def initialize(
...
-
-          @fetch_required_secrets = fetch_required_secrets
-          @resource_class = resource_class
-          @resource_id = authentication_parameters.authn_jwt_variable_id_prefix
           @http_lib = http_lib
           @create_jwks_from_http_response = create_jwks_from_http_response
+          @fetch_required_secrets = fetch_required_secrets
+          @resource_class = resource_class
+
+          @variable_id = authentication_parameters.authn_jwt_variable_id_prefix
```
Comment (orenbm) [inline app/domain/authentication/authn_jwt/signing_key/fetch_jwks_uri_signing_key.rb:25]: i think we should ditch the `@variable_id` param in all the PR and use `authentication_parameters.authn_jwt_variable_id_prefix` directly

File: app/domain/authentication/authn_jwt/signing_key/fetch_provider_uri_signing_key.rb [hunk 3]
```diff
@@ -55,23 +51,22 @@ def provider_uri_secret
         end
 
         def provider_uri_variable_id
-          @provider_uri_variable_id ||= "#{@variable_id}/#{PROVIDER_URI_RESOURCE_NAME}"
+          @provider_uri_variable_id ||= "#{@authentication_parameters.authn_jwt_variable_id_prefix}/#{PROVIDER_URI_RESOURCE_NAME}"
         end
 
         def discover_provider
```
Comment (orenbm) [inline app/domain/authentication/authn_jwt/signing_key/fetch_provider_uri_signing_key.rb:57]: this should be:
```
def discover_provider
  @logger.info(LogMessages::Authentication::AuthnJwt::FetchingJwksFromProvider.new(provider_uri))
  discovered_provider
end

def discovered_provider
  @discovered_provider ||= @discover_identity_provider.(            
    provider_uri: provider_uri          
  )
end

File: app/domain/authentication/authn_jwt/signing_key/fetch_provider_uri_signing_key.rb [hunk 4]
```diff
@@ -55,23 +51,22 @@ def provider_uri_secret
...
-        def discovered_provider
           @logger.info(LogMessages::Authentication::AuthnJwt::FetchingJwksFromProvider.new(provider_uri))
-          @discovered_provider ||= @discover_identity_provider.(
+          @discover_provider ||= @discover_identity_provider.(
             provider_uri: provider_uri
           )
         end
 
         def fetch_provider_keys
+          keys = { keys: discover_provider.jwks }
```
Comment (orenbm) [inline app/domain/authentication/authn_jwt/signing_key/fetch_provider_uri_signing_key.rb:67]:
```suggestion
          keys = { keys: discovered_provider.jwks }
```

File: app/domain/authentication/authn_jwt/validate_and_decode/fetch_issuer_value.rb [hunk 5]
```diff
@@ -121,7 +121,7 @@ def fetch_issuer_from_jwks_uri_secret
           @logger.debug(LogMessages::Authentication::AuthnJwt::ParsingIssuerFromUri.new(jwks_uri_secret_value))
 
           begin
-            @issuer_from_jwks_uri_secret ||= URI.parse(jwks_uri_secret_value).hostname
+            @issuer_from_jwks_uri_secret = URI.parse(jwks_uri_secret_value).hostname
```
Comment (orenbm) [inline app/domain/authentication/authn_jwt/validate_and_decode/fetch_issuer_value.rb:124]: `URI` can be a dependency

File: app/domain/authentication/authn_jwt/validate_and_decode/get_verification_option_by_jwt_claim.rb [hunk 6]
```diff
@@ -55,6 +54,13 @@ def get_verification_option_by_jwt_claim
 
           @verification_option
         end
+        def claim_value
```
Comment (orenbm) [inline app/domain/authentication/authn_jwt/validate_and_decode/get_verification_option_by_jwt_claim.rb:57]: please add an empty line before this

File: app/domain/authentication/authn_jwt/signing_key/create_jwks_from_http_response.rb (hunk 1 above)
Comment (tzheleznyak) [reply app/domain/authentication/authn_jwt/signing_key/create_jwks_from_http_response.rb:51]: Done

File: app/domain/authentication/authn_jwt/signing_key/fetch_jwks_uri_signing_key.rb (hunk 2 above)
Comment (tzheleznyak) [reply app/domain/authentication/authn_jwt/signing_key/fetch_jwks_uri_signing_key.rb:25]: Done

File: app/domain/authentication/authn_jwt/validate_and_decode/validate_and_decode_token.rb [hunk 7]
```diff
@@ -121,11 +117,11 @@ def verification_options_with_claims
...
         def validate_token_with_claims
-          decoded_token_after_claims_validation
+          decode_and_validate_token_with_claims
         end
 
-        def decoded_token_after_claims_validation
-          @decoded_token_after_claims_validation ||= decoded_token(verification_options_with_claims)
+        def decode_and_validate_token_with_claims
+          @decode_and_validate_token_with_claims ||= decoded_token(verification_options_with_claims)
         end
```
Comment (orenbm) [inline app/domain/authentication/authn_jwt/validate_and_decode/validate_and_decode_token.rb:125]:
```suggestion
        def decoded_and_validated_token_with_claims
          @decoded_and_validated_token_with_claims ||= decoded_token(verification_options_with_claims)
        end
```

Comment (orenbm) [reply app/domain/authentication/authn_jwt/validate_and_decode/validate_and_decode_token.rb:125]: note that in ruby we call a method that returns an object by the name of the returned object

File: app/domain/authentication/authn_jwt/validate_and_decode/validate_and_decode_token.rb [hunk 8]
```diff
@@ -24,10 +24,10 @@ def call
           validate_signature
           fetch_jwt_claims_to_validate
           validate_claims
-          decoded_token_after_claims_validation
+          decode_and_validate_token_with_claims
```
Comment (orenbm) [inline app/domain/authentication/authn_jwt/validate_and_decode/validate_and_decode_token.rb:27]: do we need to have this call? the `validate_claims` method already calls it.

File: app/domain/authentication/authn_jwt/vendor_configurations/configuration_jwt_generic_vendor.rb [hunk 9]
```diff
@@ -4,11 +4,22 @@ module VendorConfigurations
...
       class ConfigurationJWTGenericVendor < ConfigurationInterface
 
-        def initialize(authenticator_input)
+        def initialize(
+          authenticator_input:,
+          authentication_parameters_class: Authentication::AuthnJwt::AuthenticationParameters,
+          restriction_validator_class: Authentication::AuthnJwt::RestrictionValidators::ValidateRestrictionsOneToOne,
+          validate_and_decode_token_class:  Authentication::AuthnJwt::ValidateAndDecode::ValidateAndDecodeToken,
+          validate_resource_restrictions_class: Authentication::ResourceRestrictions::ValidateResourceRestrictions
+
```
Comment (orenbm) [inline app/domain/authentication/authn_jwt/vendor_configurations/configuration_jwt_generic_vendor.rb:13]: remove this empty line

File: app/domain/authentication/authn_jwt/vendor_configurations/configuration_jwt_generic_vendor.rb [hunk 10]
```diff
@@ -4,11 +4,22 @@ module VendorConfigurations
...
+        )
           @logger = Rails.logger
```
Comment (orenbm) [inline app/domain/authentication/authn_jwt/vendor_configurations/configuration_jwt_generic_vendor.rb:15]: this should go in the method signature

File: app/domain/authentication/authn_jwt/signing_key/fetch_provider_uri_signing_key.rb (hunk 3 above)
Comment (tzheleznyak) [reply app/domain/authentication/authn_jwt/signing_key/fetch_provider_uri_signing_key.rb:57]: Done

File: app/domain/authentication/authn_jwt/vendor_configurations/configuration_jwt_generic_vendor.rb [hunk 11]
```diff
@@ -80,22 +91,10 @@ def fetch_signing_key
           )
         end
 
-        def authentication_parameters_class
-          @authentication_parameters_class ||= Authentication::AuthnJwt::AuthenticationParameters
-        end
-
         def extract_token_from_credentials
           @extract_token_from_credentials ||= Authentication::AuthnJwt::InputValidation::ExtractTokenFromCredentials.new
         end
```
Comment (orenbm) [inline app/domain/authentication/authn_jwt/vendor_configurations/configuration_jwt_generic_vendor.rb:96]: this should be moved to the method signature

File: app/domain/authentication/authn_jwt/vendor_configurations/configuration_jwt_generic_vendor.rb [hunk 12]
```diff
@@ -80,22 +91,10 @@ def fetch_signing_key
...
 
-        def validate_and_decode_token_class
-          @validate_and_decode_token_class ||= Authentication::AuthnJwt::ValidateAndDecode::ValidateAndDecodeToken
-        end
-
-        def validate_resource_restrictions_class
-          @validate_resource_restrictions_class ||= Authentication::ResourceRestrictions::ValidateResourceRestrictions
-        end
-
         def create_identity_provider
```
Comment (orenbm) [inline app/domain/authentication/authn_jwt/vendor_configurations/configuration_jwt_generic_vendor.rb:98]: this should be moved to the method signature. also - i don't see why we need to have these log messages

File: app/domain/authentication/authn_jwt/signing_key/fetch_provider_uri_signing_key.rb (hunk 4 above)
Comment (tzheleznyak) [reply app/domain/authentication/authn_jwt/signing_key/fetch_provider_uri_signing_key.rb:67]: Done

File: app/domain/authentication/authn_jwt/vendor_configurations/configuration_jwt_generic_vendor.rb (hunk 12 above)
Comment (orenbm) [reply app/domain/authentication/authn_jwt/vendor_configurations/configuration_jwt_generic_vendor.rb:98]: please move the following to the method signature:
- create_signing_key_interface
- fetch_jwt_claims_to_validate
- restrictions_from_annotations

maybe also `validate_resource_restrictions`

File: app/domain/authentication/resource_restrictions/extract_resource_restrictions.rb [hunk 13]
```diff
@@ -7,7 +7,7 @@ module ResourceRestrictions
       dependencies: {
         resource_restrictions_class: Authentication::ResourceRestrictions::ResourceRestrictions,
         get_restriction_from_annotation: Authentication::ResourceRestrictions::GetRestrictionFromAnnotation.new,
-        fetch_resource_annotations: Authentication::ResourceRestrictions::FetchResourceAnnotations.new,
+        fetch_resource_annotations_instance: Authentication::ResourceRestrictions::FetchResourceAnnotations.new,
```
Comment (orenbm) [inline app/domain/authentication/resource_restrictions/extract_resource_restrictions.rb:10]: we don't need the `_instance` suffix. we don't do it anywhere else

**Review by orenbm — CHANGES_REQUESTED** [review]
left comments

File: app/domain/authentication/authn_jwt/validate_and_decode/fetch_issuer_value.rb (hunk 5 above)
Comment (tzheleznyak) [reply app/domain/authentication/authn_jwt/validate_and_decode/fetch_issuer_value.rb:124]: Done.

File: app/domain/authentication/authn_jwt/validate_and_decode/get_verification_option_by_jwt_claim.rb (hunk 6 above)
Comment (tzheleznyak) [reply app/domain/authentication/authn_jwt/validate_and_decode/get_verification_option_by_jwt_claim.rb:57]: Done

File: app/domain/authentication/authn_jwt/validate_and_decode/validate_and_decode_token.rb (hunk 7 above)
Comment (tzheleznyak) [reply app/domain/authentication/authn_jwt/validate_and_decode/validate_and_decode_token.rb:125]: Done

File: app/domain/authentication/authn_jwt/validate_and_decode/validate_and_decode_token.rb (hunk 8 above)
Comment (tzheleznyak) [reply app/domain/authentication/authn_jwt/validate_and_decode/validate_and_decode_token.rb:27]: Removed

File: app/domain/authentication/authn_jwt/vendor_configurations/configuration_jwt_generic_vendor.rb (hunk 9 above)
Comment (tzheleznyak) [reply app/domain/authentication/authn_jwt/vendor_configurations/configuration_jwt_generic_vendor.rb:13]: Done

File: app/domain/authentication/authn_jwt/vendor_configurations/configuration_jwt_generic_vendor.rb (hunk 10 above)
Comment (tzheleznyak) [reply app/domain/authentication/authn_jwt/vendor_configurations/configuration_jwt_generic_vendor.rb:15]: Done

File: app/domain/authentication/authn_jwt/vendor_configurations/configuration_jwt_generic_vendor.rb (hunk 11 above)
Comment (tzheleznyak) [reply app/domain/authentication/authn_jwt/vendor_configurations/configuration_jwt_generic_vendor.rb:96]: Done

File: app/domain/authentication/authn_jwt/vendor_configurations/configuration_jwt_generic_vendor.rb (hunk 12 above)
Comment (tzheleznyak) [reply app/domain/authentication/authn_jwt/vendor_configurations/configuration_jwt_generic_vendor.rb:98]: Moved to dependency. 
@nessiLahav take it about the logs

File: app/domain/authentication/resource_restrictions/extract_resource_restrictions.rb (hunk 13 above)
Comment (tzheleznyak) [reply app/domain/authentication/resource_restrictions/extract_resource_restrictions.rb:10]: Becaouse name fetch_resource_annotations is caught

Comment (orenbm) [reply app/domain/authentication/resource_restrictions/extract_resource_restrictions.rb:10]: by whom? can you please point me to it?

**Pull request merged**